junos/func_resource_interface_physical.go

```diff
@@ -0,0 +1,77 @@
+package junos
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+func setIntEsi(setPrefix string, esiParams []interface{},
+	m interface{}, jnprSess *NetconfObject) error {
+	sess := m.(*Session)
+	configSet := make([]string, 0)
+
+	for _, v := range esiParams {
+		if v != nil {
+			m := v.(map[string]interface{})
+			if m["identifier"].(string) != "" {
+				configSet = append(configSet, setPrefix+"esi "+m["identifier"].(string))
+			}
+			if m["mode"].(string) != "" {
+				configSet = append(configSet, setPrefix+"esi "+m["mode"].(string))
+			}
+			if m["auto_derive_lacp"].(bool) {
+				configSet = append(configSet, setPrefix+"esi auto-derive lacp")
+			}
+			if m["df_election_type"].(string) != "" {
+				configSet = append(configSet, setPrefix+"esi df-election-type "+m["df_election_type"].(string))
+			}
+			if m["source_bmac"].(string) != "" {
+				configSet = append(configSet, setPrefix+"esi source-bmac "+m["source_bmac"].(string))
+			}
+		}
+	}
+	if len(configSet) > 0 {
+		err := sess.configSet(configSet, jnprSess)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
```

Comment:
Useless if, you can simplify 
```suggestion

	return sess.configSet(configSet, jnprSess)
```